Autolcicking on holding mouse and swap hands fix

Razharas opened the pull request:
Added preference setting that makes you able to just hold down mouse
button and if you have something in your hands it will autoclick on
everything you are pointing at, no need to clickspam anymore, take care
of your fingers and mouses

Made darkness clickable, so you cna shoot at it and swap hands while clicking at it with middle mouse

Thats it

Razharas committed:
Autoclick on holding button and swap hands fix

Added preference setting that makes you able to just hold down mouse
button and if you have something in your hands it will autoclick on
everything you are pointing at, no need to clickspam anymore, take care
of your fingers and mouses
Made it so you can swap hands with middle mouse button even if clicking
on the empty void of nothingness, thus fixing some old issue i cant care
to find
Thats it

Razharas committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into TrueAuto

Iamgoofball commented:
oh shit

MrPerson commented:
Does autoclick REALLY need to be a client pref? Who the fuck is going to turn it off?

Razharas commented:
People who hate coders

MrPerson commented on code/modules/client/client procs.dm:
```diff
@@ -20,6 +20,34 @@
+		spawn()
+			while(C.is_mouse_down == object)
+				C.ClickOn(C.is_mouse_down, params)
+				sleep(1)
```
This should be sleep(world.tick_lag) so it'll work correctly if we move to 20+ fps.

Razharas replied:
It actually should be AUTOCLICK_DELAY define which i made but forgot to use

Iamgoofball commented:
for people bitching "m-m-muh autoharm-batoning!!!"

this is the same as if I had just spammed clicks manually

all this does is save me from getting carpal tunnel syndrome

@MrPerson When I'm trying to do fine precision shit and don't want to risk accidentally clicking multiple times, yes, turning it off would be nice. But, for combat, yeah, having this is beautiful.

MrPerson commented on code/modules/client/client procs.dm:
```diff
@@ -20,6 +20,34 @@
+
+/client/MouseDrag(src_object,over_object,src_location,over_location,src_control,over_control,params)
+	if(over_object)
+		MouseDown(over_object, over_location, over_control, params)
```
Is this really what we want? This'll totally mess up any other MouseDrag() actions

Razharas replied:
I know, it only works if you have something in hands, so if your hand is free its as usual
Not much can be done about it really, no way to know if you want to pull someones icon onto yourself you beat them repeatedly

MrPerson commented:
@Iamgoofball What kind of precision things? I feel we should make those things less shitty. There's also no in-game way to toggle the pref and even if there was, fix the damn problem rather than working around it!

Razharas commented:
Can add the pref change into prefs tab

vista- commented:
If this gets in, it's gonna change robust as we know it forever.

Aranclanos commented:
>client preference

why

Razharas commented:
What else?
Clearly this being the only way is not good since all kinds of drag and drop iteractions

PKPenguin321 commented:
>Made it so you can swap hands with middle mouse button even if clicking
on the empty void of nothingness

i love you

JJRcop commented:
What about if you double click and hold? Then it shows you're very intent on autolcicking, so the pref might not even be necessary, and it solves the drag n' drop conflict.

Aranclanos commented:
its awful don't do this, use the obj/screen I already told how to do it

Dorsisdwarf commented:
Is autoclick delay the same as regular clickdelay?

Razharas commented:
You attempt to click at frame rate speed but once you did hit something usual attack delay happens

freerealestate commented:
I love you

Razharas commented:
It can but:
1 - i want to leave a method to quickly get rid of the autoclick by just dropping hand
2 - this will boost weapons of any kind against unarmed strongly, with i like, disarm is too good

Razharas committed:
Made swap hands use screen object

Made swap hands use screen object
Also now blindness is restricting your ability to click
You can shoot at darkness now, it calculates the turf you want to shoot
Added toggleable verb into the prefs tab for autoclick
Made proc that calculates turf from screen loc and origin turf

optimumtact commented:
:-1: get good

Razharas committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into TrueAuto

Razharas committed:
Forgot to remove this

Forgot to remove this

RemieRichards commented on code/__HELPERS/unsorted.dm:
```diff
@@ -1304,3 +1304,14 @@ var/list/WALLITEMS = list(
+	tY = tY[1]
+	tX = text2list(tX[1], ":")
+	tX = tX[1]
+	tX = max(1, min(256, origin.x + (text2num(tX) - (world.view + 1))))
```
256 smells like something that should be world.maxX and world.maxY.

Razharas committed:
World max x instead of 256

World max x instead of 256

Aranclanos commented:
I love the autoclick for construction, it makes it stupidly faster for things like reflooring, there are some issues for when using the same object twice, like metal rods in a space turf, but its not a big deal.

Buuuuut at the same time it completely breaks combat, by changing how its played and almost securing a hit each time you run close to someone. This ends up with combat taking less time and without enough space to use the surroundings, which I believe makes the experience more enjoyable, but I guess that's just an opinion.


I don't know why the swap hands fix is tied to this pr, but you're using a screen obj that should be merged with the other two and make them be just overlays, making this future change harder and still using mouseup. Add a new screen obj.

Razharas commented:
There are a lot of screen manipulations in the code everywhere, i tried adding new object directly in the new of client but it gets cut out somewhere and im not sure it wont be cut out again by some event
I really dont want to rewrite client screen handling

Razharas committed:
Made special object instead of using blindscreen

Made special object instead of using blindscreen

Razharas committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into TrueAuto

kingofkosmos commented:
:+1: Yes please to autoclick in combat so the gap between "robust" <0.1 ping and non-american >0.2 ping players can be made smaller.

Razharas committed:
Fixing merge conflicts

Fixing merge conflicts

Aranclanos commented:
no matter how much razh asks, don't merge this, autoclick on fights is shit

Razharas commented:
Ahahaha good joke aran you got me

MrStonedOne commented:
no no no no no no no.....


There is gonna be too many accidental gun fires when people meant to do other actions.

This is just gonna end up being a nightmare

freerealestate commented:
nope

On 10 June 2015 at 15:51, MrStonedOne <notifications@github.com> wrote:

> no no no no no no no.....
>
> There is gonna be too many accidental gun fires when people meant to do
> other actions.
>
> This is just gonna end up being a nightmare
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/9765#issuecomment-110577935>
> .
>

Cheridan commented on tgstation.dme:
```diff
@@ -6,6 +6,52 @@
+#define FILE_DIR "icons"
+#define FILE_DIR "icons/ass"
+#define FILE_DIR "icons/effects"
+#define FILE_DIR "icons/mecha"
```
wowwww

Razharas replied:
Fuck this shit

Iamgoofball replied:
10/10

Razharas commented:
You can turn it off
You can drop anything in hand

Razharas committed:
Fixing accidental DME

Fixing accidental DME

vista- commented:
@MrStonedOne, that only makes this change even better. Now security can be proper incompetent.

optimumtact commented:
Please merge this to stop raz merge begging

Razharas commented:
Oranges will donate 10$ to anyone who will merge this

Aranclanos commented:
IM PRETTY SURE THATS CORRPUTION RAZH

Iamgoofball commented:
IM PRETTY SURE THE MONEY WILL JUST HAPPEN TO BE GIVEN TO THE PERSON WHO MERGES IT COMPLETELY UNRELATED TO THE MERGE ARAN

Razharas commented:
Theres literally nothing wrong with corruption
Putin told me so

kingofkosmos commented:
Let's just have an eternal test period with this like any other controversial pr.

Razharas commented:
^This

Aranclanos commented:
ok set up a test with mso using this branch instead of merging on master

Razharas committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into TrueAuto

Jordie0608 commented:
This will probably fix #5386

Razharas commented:
It will 100%

Razharas commented:
Looks like its over

Iamgoofball commented:
redo but with disables on combat so we engineers don't get carpal tunnel

kingofkosmos commented:
what why no :grimacing:

freerealestate commented:
but why?

this is needed to give people with shit latencies a fighting chance.

On 16 June 2015 at 18:22, kingofkosmos <notifications@github.com> wrote:

> what why no [image: :grimacing:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/9765#issuecomment-112307570>
> .
>

PKPenguin321 commented:
nooo

at the very least make another PR with just the middleclick on darkness fix thing
